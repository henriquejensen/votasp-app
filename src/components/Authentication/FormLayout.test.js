import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FormLayout from './FormLayout';

describe('FormLayout', () => {
  const children = (
    <form>
      <div className="field-wrapper">
        <label htmlFor="password">Senha</label>
        <input type="password" className="input" name="password" id="password" required />
      </div>
      <button className="authentication__submit-button">Entrar</button>
    </form>
  )

  it('should match a snapshot', () => {
    const props = {
      showLoginPage: jest.fn(),
      activeTab: 'signin',
      errorMessage : '',
      hideTabs: true,
      successMessage: '',
      children
    }
    const formLayout = shallow(( <FormLayout {...props} /> ))      
    expect(toJson(formLayout)).toMatchSnapshot();
  })

  it('should show successMessage', () => {
    const props = {
      showLoginPage: jest.fn(),
      activeTab: 'signin',
      errorMessage : '',
      hideTabs: true,
      successMessage: 'Success message'
    }
    const formLayout = shallow(( <FormLayout {...props} /> ))        
    expect(toJson(formLayout)).toMatchSnapshot();
  })

  it('should show errorMessage', () => {
    const props = {
      showLoginPage: jest.fn(),
      activeTab: 'signin',
      errorMessage : 'Error message',
      hideTabs: true,
      successMessage: ''
    }
    const formLayout = shallow(( <FormLayout {...props} /> ))        
    expect(toJson(formLayout)).toMatchSnapshot();
  })

  it('should call showLoginPage with true as argument', () => {
    const props = {
      showLoginPage: jest.fn(),
      activeTab: 'signup',
      errorMessage : '',
      hideTabs: false,
      successMessage: ''
    }
    const formLayout = shallow(( <FormLayout {...props} /> ));
    formLayout.find('button').first().simulate('click');
    expect(props.showLoginPage).toHaveBeenCalledWith(true);
  })

  it('should call showLoginPage with false as argument', () => {
    const props = {
      showLoginPage: jest.fn(),
      activeTab: 'signup',
      errorMessage : '',
      hideTabs: false,
      successMessage: ''
    }
    const formLayout = shallow(( <FormLayout {...props} /> ));
    formLayout.find('button').last().simulate('click');
    expect(props.showLoginPage).toHaveBeenCalledWith(false);
  })
})
