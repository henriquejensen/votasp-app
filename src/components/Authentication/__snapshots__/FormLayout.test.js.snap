// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout should match a snapshot 1`] = `
<section
  className="authentication__form-layout"
>
  <div
    className="complete-signup__background"
  />
  <div
    className="authentication__form-wrapper"
  >
    <header
      className="authentication__form-header"
    >
      <Link
        replace={false}
        to="/app"
      >
        <img
          alt="Logotipo escrito Vota em branco e SP em amarelo."
          className="logo"
          src="test-file-stub"
        />
      </Link>
    </header>
    <div
      className="authentication__tab-buttons"
    />
    <form>
      <div
        className="field-wrapper"
      >
        <label
          htmlFor="password"
        >
          Senha
        </label>
        <input
          className="input"
          id="password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="authentication__submit-button"
      >
        Entrar
      </button>
    </form>
  </div>
</section>
`;

exports[`FormLayout should show errorMessage 1`] = `
<section
  className="authentication__form-layout"
>
  <div
    className="complete-signup__background"
  />
  <div
    className="authentication__form-wrapper"
  >
    <header
      className="authentication__form-header"
    >
      <Link
        replace={false}
        to="/app"
      >
        <img
          alt="Logotipo escrito Vota em branco e SP em amarelo."
          className="logo"
          src="test-file-stub"
        />
      </Link>
    </header>
    <div
      className="authentication__tab-buttons"
    />
    <div
      className="authentication__alert"
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        Error message
      </div>
    </div>
  </div>
</section>
`;

exports[`FormLayout should show successMessage 1`] = `
<section
  className="authentication__form-layout"
>
  <div
    className="complete-signup__background"
  />
  <div
    className="authentication__form-wrapper"
  >
    <header
      className="authentication__form-header"
    >
      <Link
        replace={false}
        to="/app"
      >
        <img
          alt="Logotipo escrito Vota em branco e SP em amarelo."
          className="logo"
          src="test-file-stub"
        />
      </Link>
    </header>
    <div
      className="authentication__tab-buttons"
    />
    <div
      className="authentication__alert"
    >
      <div
        className="alert alert-success"
        role="alert"
      >
        Success message
      </div>
    </div>
  </div>
</section>
`;
